clustering/marathon.py

```diff
@@ -0,0 +1,648 @@
+#!/usr/bin/python
+# (c) 2016, Ludovic Claude <ludovic.claude@laposte.net>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = """
+module: marathon
+version_added: "2.1"
+short_description: start and stop applications with Marathon
+description:
+  - Start and stop applications with Marathon.
+author: "Ludovic Claude (@ludovicc)"
+
+options:
+  uri:
+    required: true
+    description:
+      - Base URI for the Marathon instance
+
+  state:
+    choices: [ present, absent, restart, kill ]
+    default: "present"
+    description:
+      - The operation to perform.
+
+  username:
+    required: false
+    default: null
+    description:
+      - The username to log-in with.
+
+  password:
+    required: false
+    default: null
+    description:
+      - The password to log-in with.
+
+  id:
+    required: true
+    description:
+      - Unique identifier for the app consisting of a series of names separated by slashes.
+
+  cmd:
+    aliases: [ command ]
+    required: false
+    default: null
+    description:
+      - The command that is executed.
+
+  args:
+    aliases: [ arguments ]
+    required: false
+    default: null
+    description:
+      - An array of strings that represents an alternative mode of specifying the command to run.
+
+  cpus:
+    required: false
+    default: 1.0
+    description:
+      - The number of CPU`s this application needs per instance. This number does not have to be integer, but can be a fraction.
+
+  memory:
+    aliases: [ mem ]
+    required: false
+    default: 128.0
+    description:
+      - The amount of memory in MB that is needed for the application per instance.
+
+  disk:
+    required: false
+    default: 0.0
+    description:
+      - The amount of disk in MB that is reserved for the application per instance.
+
+  ports:
+    required: false
+    default: '[]'
+    description:
+      - An array of required port resources on the host.
```

Comment:
Does this need the following adding:

An array of required port resources on the host. Required if C(require_ports=true).